nexus-repository-conan/src/main/java/org/sonatype/repository/conan/upgrade/ConanUpgrade_1_1.java

```diff
@@ -0,0 +1,156 @@
+      .type("bucket")
+      .property(P_REPOSITORY_NAME)
+      .build();
+
+  private static final String ASSET_CLASS_NAME = "asset";
+
+  private final Provider<DatabaseInstance> configDatabaseInstance;
+
+  private final Provider<DatabaseInstance> componentDatabaseInstance;
+
+  @Inject
+  public ConanUpgrade_1_1(
+      @Named(DatabaseInstanceNames.CONFIG) final Provider<DatabaseInstance> configDatabaseInstance,
+      @Named(DatabaseInstanceNames.COMPONENT) final Provider<DatabaseInstance> componentDatabaseInstance)
+  {
+    this.configDatabaseInstance = checkNotNull(configDatabaseInstance);
+    this.componentDatabaseInstance = checkNotNull(componentDatabaseInstance);
+  }
+
+  @Override
+  public void apply() {
+    if (hasSchemaClass(configDatabaseInstance, "repository") &&
+        hasSchemaClass(componentDatabaseInstance, ASSET_CLASS_NAME)) {
+      List<String> repositoryNames = findConanRepositoryNames();
+      updateAssetPath(repositoryNames);
+      removeAttributesFromConanManifest(repositoryNames);
+    }
+  }
+
+  private List<String> findConanRepositoryNames() {
+    try (ODatabaseDocumentTx oDatabaseDocumentTx = configDatabaseInstance.get().connect()) {
+      final List<ODocument> documents = oDatabaseDocumentTx.query(
+          new OSQLSynchQuery<ODocument>("select from repository where recipe_name in ['conan-proxy', 'conan-hosted']"));
+      return documents
+          .stream()
+          .map(entries -> (String) entries.field(P_REPOSITORY_NAME))
+          .collect(Collectors.toList());
+    }
+  }
+
+  private void removeAttributesFromConanManifest(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> repositoryNames.forEach(repositoryName -> {
+          OIndex<?> bucketIdx = db.getMetadata().getIndexManager().getIndex(I_REPOSITORY_NAME);
+          OIdentifiable bucket = (OIdentifiable) bucketIdx.get(repositoryName);
+
+          List<ODocument> conanManifests = db.query(new OSQLSynchQuery<ODocument>(
+                  "select from asset where bucket = ? and attributes.conan.asset_kind = 'CONAN_MANIFEST'"),
+              bucket.getIdentity());
+
+          conanManifests
+              .forEach(oDocument -> {
+
+                Map<String, Object> attributes = oDocument.field("attributes");
+                // remove all attributes, except asset_kind from conan "bucket"
+                attributes.put(ConanFormat.NAME, Collections.singletonMap(P_ASSET_KIND, AssetKind.CONAN_MANIFEST.name()));
+                oDocument.field("attributes", attributes);
+
+                oDocument.save();
+              });
+        }));
+  }
+
+  private void updateAssetPath(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> repositoryNames
+            .stream()
+            .flatMap(repositoryName -> {
```

Comment:
might align this method with the `removeAttributesFromConanManifest` method, in that you could use this flatMap logic in both instances (or alternatively don't use the flatMap on the stream in this method, and simply do a forEach on repositoryNames as done in the other method, don't particularly care which, but should be consistent)

Reply:
was created one method called findAssets
fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/95/commits/9938f7c67cac959dbbf6292c9271250ee6812827